### PR TITLE
CI Error on FutureWarning

### PR DESCRIPTION
**Pull request by adrinjalali**
This PR makes the CI to fail on `FutureWarning` so that our test suite won't use our own deprecated API and our codebase use new API from our dependencies.

### CODE REVIEW

**Comment by adrinjalali** [comment]
Would be nice to have this in so that we can merge https://github.com/huggingface/huggingface_hub/pull/745 after

**Review by LysandreJik — APPROVED** [review]
This looks good to me, thanks for working on it @adrinjalali!

File: tests/test_snapshot_download.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ def setUpClass(cls):
         """
         Share this valid token in all tests below.
         """
-        cls._token = cls._api.login(username=USER, password=PASS)
+        cls._token = TOKEN
+        cls._api.set_access_token(TOKEN)
```
Comment (osanseviero) [inline tests/test_snapshot_download.py:31]: Do we really want to do this? This means that if someone runs the test locally, they will be changing their credential store no?

**Review by osanseviero — COMMENTED** [review]
Looks great, thanks for this change! I left a small question

File: tests/test_snapshot_download.py (hunk 1 above)
Comment (julien-c) [reply tests/test_snapshot_download.py:31]: i think that's already the case anyways for other tests no?

Comment (adrinjalali) [reply tests/test_snapshot_download.py:31]: Yes it does, but that's already the case (which I think shouldn't be). I'll open a separate issue to discuss that.

Comment (osanseviero) [reply tests/test_snapshot_download.py:31]: I don't think that's true. `login` does not update the git credential store, it just gives you the token.

Comment (adrinjalali) [reply tests/test_snapshot_download.py:31]: `login` stores `username` and `password`, just not the token. I don't think either of them should: https://github.com/huggingface/huggingface_hub/issues/796

```python
    def login(self, username: str, password: str) -> str:
        """
        Call HF API to sign in a user and get a token if credentials are valid.

        Outputs: token if credentials are valid

        Throws: requests.exceptions.HTTPError if credentials are invalid
        """
        warnings.warn(
            "HfApi.login: This method is deprecated in favor of `set_access_token`"
            " and will be removed in v0.7.",
            FutureWarning,
        )
        path = f"{self.endpoint}/api/login"
        r = requests.post(path, json={"username": username, "password": password})
        r.raise_for_status()
        d = r.json()

        write_to_credential_store(username, password)
        return d["token"]

    def set_access_token(access_token: str):
        write_to_credential_store(USERNAME_PLACEHOLDER, access_token)

```

Comment (osanseviero) [reply tests/test_snapshot_download.py:31]: Ah thank you for pointing it out!

**Pull request merged**